src/qcodes/parameters/parameter_base.py

```diff
@@ -1144,6 +1160,82 @@ def underlying_instrument(self) -> InstrumentBase | None:
     def abstract(self) -> bool | None:
         return self._abstract
 
+    @property
+    def param_spec(self) -> ParamSpecBase:
+        match self.vals:
+            case Arrays():
+                paramtype = "array"
+            case Strings():
+                paramtype = "text"
+            case ComplexNumbers():
+                paramtype = "complex"
+            case _:
+                paramtype = "numeric"
+
+        return ParamSpecBase(
+            name=self.register_name,
+            paramtype=paramtype,
```

Comment:
In the long run I think we should change the name of this property to something like sqlite_storage_class ... but probably not in this pr

Reply:
I agree it would be nice to have a more descriptive name, and also that it is beyond the scope of this PR.